scipy/optimize/_zeros_py.py

```diff
@@ -1392,3 +1392,125 @@ def toms748(f, a, b, args=(), k=1,
                           maxiter=maxiter, disp=disp)
     x, function_calls, iterations, flag = result
     return _results_select(full_output, (x, function_calls, iterations, flag))
+
+
+def _bracketer(f, a, b=None, min=None, max=None, factor=2, batch=100):
+    """Bracket the root of a monotonic scalar function of one variable
+
+    Parameters
+    ----------
+    f : callable
+        The function for which the root is to be bracketed.
+    a, b : float
+        Starting guess of bracket, which need not contain a root. If `b` is
+        not provided, ``b = a + 1``.
+    min, max : float
+        Minimum and maximum allowable endpoints of the bracket, inclusive.
+    factor : float
+        The factor used to grow the bracket. See notes for details.
```

Comment:
This could also be a function.

Also, I am thinking that it could use a similar logic as simulated annealing with a temperature following an exponential profile.

Reply:
Could be. With this PR, I only intended to reproduce (in the form of a function) the algorithms we already have, not create a new algorithm.